application/actions/auth.go

```diff
@@ -235,51 +180,111 @@ func createAuthUser(
 	return authUser, nil
 }
 
-func AuthLogin(c buffalo.Context) error {
-	clientID, err := getOrSetClientID(c)
-	if err != nil {
-		return err
+func AuthRequest(c buffalo.Context) error {
+	clientID := c.Param(ClientIDParam)
+	if clientID == "" {
+		return authError(c, http.StatusBadRequest, "MissingClientID", ClientIDParam+" is required to login")
+	}
+
+	c.Session().Set(ClientIDSessionKey, clientID)
+
+	authEmail := c.Param(AuthEmailParam)
+	if authEmail == "" {
+		return authError(c, http.StatusBadRequest, "MissingAuthEmail", AuthEmailParam+" is required to login")
 	}
+	c.Session().Set(AuthEmailSessionKey, authEmail)
+
+	getOrSetReturnTo(c)
```

Comment:
Why is this handled differently than client ID and auth email? I think it would read easier to directly set the returnTo session key and rename `getOrSetReturnTo` to `getReturnTo`. Unless I'm missing something.

Reply:
The main reason I left it unchanged was because it never returns an error - it's more flexible by just returning "/#" if it doesn't find anything.